spec/field_types.md

```diff
@@ -11,10 +11,27 @@ _DigestSet (object)_
 > It is RECOMMENDED to use at least `sha256` for compatibility between producers
 > and consumers.
 >
-> Pre-defined algorithm names: `sha256`, `sha224`, `sha384`, `sha512`,
-> `sha512_224`, `sha512_256`, `sha3_224`, `sha3_256`, `sha3_384`, `sha3_512`,
-> `shake128`, `shake256`, `blake2b`, `blake2s`, `ripemd160`, `sm3`, `gost`,
-> `sha1`, `md5`.
+> Pre-defined algorithms:
+>
+> -   `sha256`, `sha224`, `sha384`, `sha512`, `sha512_224`, `sha512_256`,
+>     `sha3_224`, `sha3_256`, `sha3_384`, `sha3_512`, `shake128`, `shake256`,
+>     `blake2b`, `blake2s`, `ripemd160`, `sm3`, `gost`, `sha1`, `md5`
+>
+>     Standard cryptographic hash algorithms, for cases when when the method of
+>     serialization is obvious or well known.
+>
+> -   `goModuleH1`: The go module [directory Hash1], omitting the "h1:"
+>     prefix and output in hexadecimal instead of base64. Can be computed over a
+>     directory named `name@version`, or the contents of zip file containing
+>     such a directory:
+>
+>     ```bash
+>     find name@version -type f | LC_ALL=C sort | xargs -r sha256sum | sha256sum | cut -f1 -d' '
```

Comment:
Is this the actual command to use in order to generate the goModuleH1 value on a Go module directory? 

Sorry if this is a stupid question, I'm trying to add it into our provenance generation and as a sanity check I'm running it on different copies of the same source to validate that the value is the same (i.e. I would expect the hash to be the same if the file contents are the same). 

Actually it gives a different hash value depending on the folder path passed into the command - am I using it wrong?

i.e.

khalk@khalk:~/tempgit$ git clone https://github.com/golang/example
khalk@khalk:~/tempgit$ find ~/tempgit/example -type f | LC_ALL=C sort | xargs -r sha256sum | sha256sum | cut -f1 -d' '
45bde95411f095cd1040f6e5ad0c3f25f3c7ec8b9b048a32ceef01bac110e010
khalk@khalk:~/tempgit$ cd example
khalk@khalk:~/tempgit/example$ find . -type f | LC_ALL=C sort | xargs -r sha256sum | sha256sum | cut -f1 -d' '
51d525520df583ea4c543d00381eb197c4190e39d92ff37feee1ee048a92e2e6

Reply:
The problem is that you need to literally type `find name@version` because the "name@version" is part of the output. If you use "~/tempgit/example" or "." or "./name@version" you get a different result. I need to update the text to make this more clear.

Here's a script that is less error prone: https://gist.github.com/MarkLodato/c03659d242ea214ef3588f29b582be70

Reply:
The name of the directory itself does not matter. What matters is that the outpuf of the first `sha256sum` command looks like this:

```
abcd1234...90  example@v1.0/file.go
dcba4321...88  example@v1.0/another-file.go
...
```

That is, a relative path whose first component is `<name>@<version>`.

My script does a `sed` command to replace the actual directory name with the second parameter, which would be another way to achieve this.

Alternatively, you could write a small go program to call the official [HashDir](https://cs.opensource.google/go/x/mod/+/master:sumdb/dirhash/hash.go;l=70;drc=346a37af5599be02f125bd8bc0a5e1d33db21ddc) function.